trieste/acquisition/multi_objective/partition.py

```diff
@@ -45,35 +49,48 @@ def prepare_default_non_dominated_partition_bounds(
     :return: lower, upper bounds of the partitioned cell, each with shape [N, D]
     :raise ValueError (or `tf.errors.InvalidArgumentError`): If ``reference`` has an invalid
         shape.
+    :raise ValueError (or `tf.errors.InvalidArgumentError`): If ``anti_reference`` has an invalid
+        shape.
     """
-    assert tf.shape(observations)[-1] >= 2, ValueError(
-        "the last dimension of multi-objective "
-        f"observations (i.e., objective number) is expected to be at "
-        f"least 2 but found {tf.shape(observations)[-1]}"
-    )
-    if (
-        anti_reference is None
-    ):  # if anti_reference point is not specified, use a -1e10 as default (act as -inf)
-        # make sure given observations are larger than -1e10
-        anti_reference = -1e10 * tf.ones(
-            shape=(tf.shape(observations)[-1]), dtype=observations.dtype
-        )
-        tf.debugging.assert_greater_equal(
-            observations,
-            anti_reference,
-            message="observations containing points below default "
-            "anti-reference point ([-1e10, ..., -1e10]), try specify a lower "
-            "anti-reference point.",
-        )
+
+    def not_valid_obs(obs: Optional[TensorType]) -> bool:
+        return obs is None or tf.equal(tf.size(observations), 0)
+
+    def specify_default_anti_reference_point(
+        ref: TensorType, obs: Optional[TensorType]
+    ) -> TensorType:
+        anti_ref = -1e10 * tf.ones(shape=(tf.shape(reference)), dtype=reference.dtype)
         tf.debugging.assert_greater_equal(
-            reference,
-            anti_reference,
+            ref,
+            anti_ref,
             message="reference point containing dimensionality below default "
             "anti-reference point ([-1e10, ..., -1e10]), try specify a lower "
             "anti-reference point.",
         )
+        if not not_valid_obs(obs):  # make sure given (valid) observations are larger than -1e10
+            tf.debugging.assert_greater_equal(
+                obs,
+                anti_ref,
+                message="observations containing points below default "
+                "anti-reference point ([-1e10, ..., -1e10]), try specify a lower "
+                "anti-reference point.",
+            )
+        return anti_ref
 
-    if tf.shape(observations)[-1] > 2:
+    tf.debugging.assert_shapes([(reference, ["D"])])
+    if (  # prepare default anti_reference point if not specified
+        anti_reference is None
+    ):  # if anti_reference point is not specified, use a -1e10 as default (act as -inf)
+        anti_reference = specify_default_anti_reference_point(reference, observations)
+    else:  # anti_reference point is specified
+        tf.debugging.assert_shapes([(anti_reference, ["D"])])
+
+    if not_valid_obs(observations):  # if no valid observations
+        assert tf.reduce_all(tf.less_equal(anti_reference, reference)), ValueError(
```

Comment:
isn't this something we want to test all the time? (i.e. whether or not observations are valid or not)

Reply:
also change message to "anti_reference point contains at least one value larger than reference point" ?

Reply:
> isn't this something we want to test all the time? (i.e. whether or not observations are valid or not)

Not sure if got the point (please correct if misinterpreted), thought we only need to test when user specified a reference point (as otherwise the current automatic reference selection would always result valid `observation`).

Reply:
My point was just that we could test `anti_reference << reference` all the time, not only if `not_valid_obs(observations)`.

Anyway this is not very important.

Reply:
ah i see, was thinking as `prepare_default_non_dominated_partition_bounds` only add a "new" partition strategy for empty observations, hence we only do `anti_reference << reference` in this new partition. As in other cases, this function will invoke other partition method and its these partition methods responsibility to check if `anti_reference << reference` (i.e., [this test](https://github.com/secondmind-labs/trieste/blob/45f99cab9d0bdc5a174fdf5d742f3f52a20ae9ea/tests/unit/acquisition/multi_objective/test_partition.py#L118-L127) and [this test](https://github.com/secondmind-labs/trieste/blob/45f99cab9d0bdc5a174fdf5d742f3f52a20ae9ea/tests/unit/acquisition/multi_objective/test_partition.py#L218-L232))